Aqovia.Utilities.SagaMachine.Tests/StateMachineTests.cs

```diff
@@ -14,9 +14,27 @@ public class StateMachineTests
     {
         private readonly Mock<IKeyValueStore> _keyValueStoreMock;
         private readonly Mock<Func<IEnumerable<ISagaMessageIdentifier>, Task>> _mockPublisher;
-        private readonly SagaMachine<TestState> _sagaMachine;
+        private SagaMachine<TestState> _sagaMachine;
+        private readonly Mock<IEventLoggerFactory> _mockEventloggerFactory;
         private Mock<IEventLogger> _mockEventLogger;
 
+        private class MessagePublisherFake
+        {
+            public async Task PublishMessage(IEnumerable<ISagaMessageIdentifier> messsages)
```

Comment:
This method can be static and doesn't need a class around it. I'd also call it something that indicates that it's expected to throw, e.g. `ExceptionThrowingPublishMessage`.